File: Calculation/Leibniz-formula/C#/Leibniz.cs
using System;

class Program
{
    // ライプニッツ公式計算
    static double CalculateLeibnizFormula(int terms)
    {
        double piApprox = 0.0;
        double sign = 1.0;
        for (int i = 0; i < terms; i++)
        {
            piApprox += sign / (2.0 * i + 1.0);
            sign = -sign;
        }
        piApprox *= 4.0;
        return piApprox;
    }

    static void Main()
    {
        // 計算する項数(1億回)
        const int n = 100000000;

        // ライプニッツ公式計算
        double piApprox = CalculateLeibnizFormula(n);

        // 誤差
        double calculationError = Math.PI - piApprox;

        Console.WriteLine("円周率近似値  : {0}", piApprox);
        Console.WriteLine("計算誤差{0}   : {0}", calculationError);
    }
}

File: Calculation/Leibniz-formula/Python3/Leibniz.py
import math

# ライプニッツ公式計算
def calculate_leibniz_formula(n):
    pi_approx = 0
    sign = 1
    for i in range(n):
        pi_approx += sign / (2 * i + 1)
        sign = -sign
    pi_approx *= 4
    return pi_approx


# 計算する項数(1億回)
n = 100000000

# ライプニッツ公式計算
pi_approx = calculate_leibniz_formula(n)

# 誤差
calculation_error = math.pi - pi_approx

print("円周率近似値  : ", pi_approx)
print("計算誤差      : ", calculation_error)